service/internal/proctelemetry/process_telemetry.go

```diff
@@ -31,100 +48,224 @@ type processMetrics struct {
-	if err != nil {
-		return err
-	}
-	if err = pm.processUptime.UpsertEntry(pm.updateProcessUptime); err != nil {
-		return err
-	}
+		ocmetric.WithDescription("Uptime of the process"),
+		ocmetric.WithUnit(stats.UnitSeconds))
+	errors = multierr.Append(errors, err)
 
 	pm.allocMem, err = registry.AddInt64DerivedGauge(
 		"process/runtime/heap_alloc_bytes",
-		metric.WithDescription("Bytes of allocated heap objects (see 'go doc runtime.MemStats.HeapAlloc')"),
-		metric.WithUnit(stats.UnitBytes))
-	if err != nil {
-		return err
-	}
-	if err = pm.allocMem.UpsertEntry(pm.updateAllocMem); err != nil {
-		return err
-	}
+		ocmetric.WithDescription("Bytes of allocated heap objects (see 'go doc runtime.MemStats.HeapAlloc')"),
+		ocmetric.WithUnit(stats.UnitBytes))
+	errors = multierr.Append(errors, err)
 
 	pm.totalAllocMem, err = registry.AddInt64DerivedCumulative(
 		"process/runtime/total_alloc_bytes",
-		metric.WithDescription("Cumulative bytes allocated for heap objects (see 'go doc runtime.MemStats.TotalAlloc')"),
-		metric.WithUnit(stats.UnitBytes))
-	if err != nil {
-		return err
-	}
-	if err = pm.totalAllocMem.UpsertEntry(pm.updateTotalAllocMem); err != nil {
-		return err
-	}
+		ocmetric.WithDescription("Cumulative bytes allocated for heap objects (see 'go doc runtime.MemStats.TotalAlloc')"),
+		ocmetric.WithUnit(stats.UnitBytes))
+	errors = multierr.Append(errors, err)
 
 	pm.sysMem, err = registry.AddInt64DerivedGauge(
 		"process/runtime/total_sys_memory_bytes",
-		metric.WithDescription("Total bytes of memory obtained from the OS (see 'go doc runtime.MemStats.Sys')"),
-		metric.WithUnit(stats.UnitBytes))
-	if err != nil {
-		return err
-	}
-	if err = pm.sysMem.UpsertEntry(pm.updateSysMem); err != nil {
-		return err
-	}
+		ocmetric.WithDescription("Total bytes of memory obtained from the OS (see 'go doc runtime.MemStats.Sys')"),
+		ocmetric.WithUnit(stats.UnitBytes))
+	errors = multierr.Append(errors, err)
 
 	pm.cpuSeconds, err = registry.AddFloat64DerivedCumulative(
 		"process/cpu_seconds",
-		metric.WithDescription("Total CPU user and system time in seconds"),
-		metric.WithUnit(stats.UnitSeconds))
-	if err != nil {
-		return err
-	}
-	if err = pm.cpuSeconds.UpsertEntry(pm.updateCPUSeconds); err != nil {
-		return err
-	}
+		ocmetric.WithDescription("Total CPU user and system time in seconds"),
+		ocmetric.WithUnit(stats.UnitSeconds))
+	errors = multierr.Append(errors, err)
 
 	pm.rssMemory, err = registry.AddInt64DerivedGauge(
 		"process/memory/rss",
-		metric.WithDescription("Total physical memory (resident set size)"),
-		metric.WithUnit(stats.UnitBytes))
-	if err != nil {
-		return err
-	}
-	if err = pm.rssMemory.UpsertEntry(pm.updateRSSMemory); err != nil {
-		return err
-	}
+		ocmetric.WithDescription("Total physical memory (resident set size)"),
+		ocmetric.WithUnit(stats.UnitBytes))
+	errors = multierr.Append(errors, err)
 
-	return nil
+	return errors
+}
+
+func (pm *processMetrics) recordWithOC() error {
```

Comment:
As per the comment above, I would suggest us not changing these.

Reply:
removed!